hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/TableModificationProcedureUtils.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.master.procedure;
+
+import org.apache.hadoop.hbase.TableName;
+import org.apache.yetus.audience.InterfaceAudience;
+
+@InterfaceAudience.Private
+public class TableModificationProcedureUtils {
+
+  private TableModificationProcedureUtils() {
+  }
+
+  /**
+   * Some procedures cannot safely run while a table is being modified. Helper method to allow them
+   * to check if such a procedure is running on a table.
+   */
+  public static boolean isTableModificationInProgress(MasterProcedureEnv env, TableName tableName) {
+    return env.getMasterServices().getMasterProcedureExecutor().getProcedures().stream()
+      .filter(p -> !p.isFinished())
+      .filter(p -> (p instanceof CreateTableProcedure || p instanceof TruncateTableProcedure
+        || p instanceof ModifyTableProcedure || p instanceof EnableTableProcedure
+        || p instanceof DisableTableProcedure || p instanceof CloneSnapshotProcedure))
```

Comment:
Instead of matching on specific classes, will this be easier to reason about if we instead select on a collection of `TableOperationType`s ? That way the implementation details can be fluid while the intention remains sound.